Index: deepin-initramfs-tools/update-initramfs
===================================================================
--- deepin-initramfs-tools.orig/update-initramfs	2025-02-12 18:21:15.682113353 +0800
+++ deepin-initramfs-tools/update-initramfs	2025-02-13 11:52:16.975216439 +0800
@@ -138,15 +138,16 @@
 	if [ "${verbose}" = 1 ]; then
 		OPTS="-v ${OPTS}"
 	fi
+	new_file="${initramfs}.new.$$"
 	# shellcheck disable=SC2086
-	if mkinitramfs ${OPTS} "${initramfs}.new" "${version}"; then
-		mv -f "${initramfs}.new" "${initramfs}"
+	if mkinitramfs ${OPTS} "${new_file}" "${version}"; then
+		mv -f "${new_file}" "${initramfs}"
 		# Guard against an unclean shutdown
 		sync -f "${initramfs}"
 	else
 		mkinitramfs_return="$?"
 		remove_initramfs_bak
-		rm -f "${initramfs}.new"
+		rm -f "${new_file}"
 		echo "update-initramfs: failed for ${initramfs} with $mkinitramfs_return." >&2
 		exit $mkinitramfs_return
 	fi
