Index: deepin-initramfs-tools/scripts/local-premount/resume
===================================================================
--- deepin-initramfs-tools.orig/scripts/local-premount/resume	2025-02-12 18:21:15.682113353 +0800
+++ deepin-initramfs-tools/scripts/local-premount/resume	2025-02-13 11:21:03.969009168 +0800
@@ -26,17 +26,6 @@
 	exit 0
 fi
 
-if [ "$(get_fstype "${DEV}")" = suspend ] &&
-   command -v plymouth >/dev/null &&
-   plymouth --ping; then
-	plymouth display-message --text="Resuming from hibernation"
-
-	# The above command does not wait for a framebuffer update to
-	# complete, so the kernel can freeze the framebuffer before
-	# the message is even visible.  Wait just a moment to make
-	# that less likely.
-	sleep 0.1
-fi
 
 # hardcode path, uswsusp ships an resume binary too
 if [ -n "${resume_offset?}" ]; then
