Index: deepin-initramfs-tools/scripts/functions
===================================================================
--- deepin-initramfs-tools.orig/scripts/functions	2025-02-12 18:21:15.678113198 +0800
+++ deepin-initramfs-tools/scripts/functions	2025-05-09 19:44:22.926816821 +0800
@@ -71,6 +71,11 @@
 
 	run_scripts /scripts/panic
 
+	# The existence of this file indicates that it can be used to continue booting
+	if [ -f "/tmp/panic_continue_boot" ]; then
+		return;
+	fi
+
 	# Try to use setsid, which will enable job control in the shell
 	# and paging in more
 	if command -v setsid >/dev/null 2>&1; then
